lazyflow/operators/ioOperators/opFormattedDataExport.py

```diff
@@ -258,3 +279,36 @@ def run_export(self):
 
     def run_export_to_array(self):
         return self._opExportSlot.run_export_to_array()
+
+    def run_distributed_export(self, block_shape: Shape5D):
+        from lazyflow.distributed.TaskOrchestrator import TaskOrchestrator
+
+        orchestrator = TaskOrchestrator()
+        n5_file_path = Path(self.OutputFilenameFormat.value).with_suffix(".n5")
+        output_meta = self.ImageToExport.meta
+        if orchestrator.rank == 0:
+            output_shape = output_meta.getShape5D()
+            block_shape = block_shape.clamped(maximum=output_shape)
+
+            with z5py.File(n5_file_path, "w") as f:
+                ds = f.create_dataset(
+                    self.OutputInternalPath.value,
+                    shape=output_meta.shape,
+                    chunks=block_shape.to_tuple(output_meta.getAxisKeys()),
+                    dtype=output_meta.dtype.__name__,
+                )
+                ds.attrs["axes"] = list(reversed(output_meta.getAxisKeys()))
+                ds[...] = 1  # FIXME: for some reason setting to 0 does nothing
+
+            cutout = self.get_roi()
+            orchestrator.orchestrate(cutout.split(block_shape=block_shape))
+        else:
+
+            def process_tile(tile: Slice5D, rank: int):
+                self.set_roi(tile)
+                slices = tile.to_slices(output_meta.getAxisKeys())
+                with z5py.File(n5_file_path, "r+") as n5_file:
+                    dataset = n5_file[self.OutputInternalPath.value]
+                    dataset[slices] = self.ImageToExport.value
```

Comment:
I'm a bit concerned about this approach (please correct my if I have overlooked something here). Please correct me if I am wrong, but each worker will always process a whole block at once. With the standard block size of 256, this can blow up with 3D time series. Just consider a 2-channel block: `256**4 * 2 * 32bit ~ 35GB`. In order not to blow up nodes with this configuration one would feel inclined to lower the block size, but this will be inefficient quickly due to halo.
Blocking over time should be reconsidered IMO.

Reply:
True that; I overlooked 3D data and/or time series.... Well, the blocks can have arbitrary shape, so maybe defaulting to something more sensible, like

```
{x:256, y: 256, t: 1, z: 1, c:slice(None)}
```
would be a better idea? We could make t so that the user must specify a block size instead of trying to be clever about

Reply:
I'd keep 3D blocks 3D, 2D blocks 2D, so something like

```
{x:256, y: 256, t: 1, z: 256, c:slice(None)}
```
this would be "clamped" in the case of 2D data anyway. But I wouldn't want to force slice-wise computation